src/core/addonManager.js

```diff
@@ -164,18 +165,39 @@ export function addChangeListeners(
   mozAddonManager: MozAddonManagerType,
 ) {
   function handleChangeEvent(e: AddonChangeEvent) {
-    const { id, type, needsRestart } = e;
-
-    log.info('Event received', { type, id, needsRestart });
+    const { id: guid, type, needsRestart } = e;
+
+    log.info('Event received', { type, id: guid, needsRestart });
+
+    if (type === ON_OPERATION_CANCELLED_EVENT) {
+      // We need to retrieve the correct status for this add-on.
+      return getAddon(guid, { _mozAddonManager: mozAddonManager })
+        .then((addon) => {
+          const status = getAddonStatus({ addon });
+
+          return callback({
+            guid,
+            status,
+            needsRestart,
+          });
+        })
+        .catch((error) => {
+          log.error(
```

Comment:
This catch block is missing test coverage. I think it's important to cover this case because it won't get used a lot in real life. Yes, I know that statement sounds weird 😁